docs/development/decision-records/2024-08-28-Release-Workflow-Refactor/README.md

```diff
@@ -0,0 +1,82 @@
+- The `publish-new-release` workflow can't be manually triggered;
+- Even if it could be manually triggered, there are steps that shouldn't apply for bugfix releases such as bumping the project version or merging to main;
+
+
+## Approach
+
+The draft-new-release workflow should include one extra inputs:
+- A `choice` input to allow the distinction between a release or bugfix.
+
+Github Actions Jobs and Step conditionals are then used to make the execution follow the required steps, as follows.
+
+### Prepare Release workflow logic
+- Triggers manually. Input is the version, choice and base ref (should be either commit from main or tag)
+- if choice is release
+  - branches from main into `release/<input_version>`
+- if choice is bugfix
+  - branches from tag into `bugfix/<input_version>`
+- replaces the project default version `(gradle.properties)` with the workflow version input
+- replaces openapi spec version
+- replaces charts version and README
+- commits and pushes to `release/*` or `bugfix/X.Y.Z` branch
+- if new release
+   - opens PR with new `release/*` branch as head and `main/` as base
+
+Then, a different process takes places if we intend to release a latest version or a bugfix. 
+
+### Case for latest releases
+After the prepare release PR is approved, merging into main will trigger the publish-release workflow where the required publication steps will take place, as shown in [Publish release workflow logic](#publish-release-workflow-logic).
+
+### Case for bugfix Releases
+After the bugfix branch is created, developers commit the fix or set of fixes to it via PR.
+The release of the bugfix version should be triggered manually through the publishing workflow and should execute most of the publication steps as a normal release.
+However, for a bugfix release, the workflow shouldn't:
+- Make any commits to the main branch
+- Bump the project version to the next minor version
+
+### Publish release workflow logic
+- Triggers automatically when prepare release PR is merged into main from `release/*` branch
+- Triggers manually only from branch `bugfix/X.Y.Z`
+
+- Extracts release version from `gradle.properties`
+- Triggers release to maven repository by calling `trigger-maven-publish.yaml`
+- Triggers docker image publishing by calling `trigger-maven-publish.yaml`
+- Releases helm-charts
+- Creates Github Tag and Release
+- if PR was merged to main it means a new release then:
+   - Updates project version to next version, commits and pushes to main
+      - if final version, increments to next minor version and adds "-SNAPSHOT"
+      - if not final version (e.g.-rc1), do not increment
+- _(elif PR was merged to hotfix then don't increment)_
+- Publishes to Swaggerhub by calling `publish-swaggerhub.yaml`
+- Publishes OpenAPI UI spec to Github Pages by calling `publish-openapi-ui.yml`
+
+## Other Considerations
+
+- Keeping `release/*` branches is no longer needed since the tag references the commit on main.
```

Comment:
we actually have `release/<input_version>` branches, and a `releases/` branch. will both get :axe:'d?
Also, probably worth mentioning, that `bugfix/` branches will indeed stay, for later subsequent bugfixes.

At that point, I would not be opposed to also keep the `release/` branches around, simply for consistency's sake.

Reply:
I'd remove the `release/<input_version>` as well since we have the tags. I see no reason to keep both.